src/v/cluster/topics_frontend.cc

```diff
@@ -353,24 +383,29 @@ ss::future<std::vector<topic_result>> topics_frontend::delete_topics(
 ss::future<topic_result> topics_frontend::do_delete_topic(
   model::topic_namespace tp_ns, model::timeout_clock::time_point timeout) {
     delete_topic_cmd cmd(tp_ns, tp_ns);
+    try {
+        // Delete data-policy
+        auto delete_dp = co_await _dp_frontend.local().clear_data_policy(
+          tp_ns, timeout);
+        if (
+          delete_dp != std::error_code(cluster::errc::success)
+          && delete_dp
+               != std::error_code(cluster::errc::data_policy_not_exists)) {
+            co_return topic_result(std::move(tp_ns), map_errc(delete_dp));
+        }
```

Comment:
I sort of think we should wait to add this functionality in a later PR. To me it isn't clear that we should delete a data policy when a topic is deleted. The policy of deleting the data policy when a topic is deleted shouldn't affect the initial set of PRs and then we can discuss this topic deletion policy separately.